Pinta/Dialogs/CanvasGridSettingsDialog.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using Gtk;
+using Pinta.Core;
+using Pinta.Core.Managers;
+
+namespace Pinta;
+
+public sealed class CanvasGridSettingsDialog : Dialog
+{
+	private readonly CanvasGridManager canvas_grid;
+
+	private readonly bool initial_show_grid;
+	private readonly int initial_grid_width;
+	private readonly int initial_grid_height;
+
+	private readonly CheckButton show_grid_checkbox;
+	private readonly SpinButton grid_width_spinner;
+	private readonly SpinButton grid_height_spinner;
+
+	private const int Spacing = 6;
+
+	internal CanvasGridSettingsDialog (
+		ChromeManager chrome,
+		CanvasGridManager canvasGrid)
+	{
+		canvas_grid = canvasGrid;
+
+		initial_show_grid = canvas_grid.ShowGrid;
+		initial_grid_width = canvas_grid.CellWidth;
+		initial_grid_height = canvas_grid.CellHeight;
+
+		show_grid_checkbox = CreateShowGridCheckbox (initial_show_grid, SettingsChanged);
+		grid_width_spinner = CreateSpinner (initial_grid_width, SettingsChanged);
+		grid_height_spinner = CreateSpinner (initial_grid_height, SettingsChanged);
+
+		Grid grid = new () {
+			RowSpacing = Spacing,
+			ColumnSpacing = Spacing,
+			ColumnHomogeneous = false,
+		};
+
+		grid.Attach (show_grid_checkbox, 0, 0, 2, 1);
+
+		grid.Attach (CreateLabel ("Width:", Align.End), 0, 1, 1, 1);
+		grid.Attach (grid_width_spinner, 1, 1, 1, 1);
+		grid.Attach (Label.New (Translations.GetString ("pixels")), 2, 1, 1, 1);
+
+		grid.Attach (CreateLabel ("Height:", Align.End), 0, 2, 1, 1);
+		grid.Attach (grid_height_spinner, 1, 2, 1, 1);
+		grid.Attach (Label.New (Translations.GetString ("pixels")), 2, 2, 1, 1);
+
+		Box mainVbox = new () { Spacing = Spacing };
+		mainVbox.SetOrientation (Orientation.Vertical);
+		mainVbox.Append (grid);
+
+		Title = Translations.GetString ("Grid Settings");
```

Comment:
The dialog title could be `Canvas Grid Settings` to match the naming of the menu label suggested in my other comment?